### PR TITLE
User functionalities

### PR DESCRIPTION
**Pull request by varnita619**
- added edit profile functionality
- follow/unfollow user
- sort by trending, sort by date
- display user feed
- display users post in user Profile page

### CODE REVIEW

**Review by samyakshah3008 — APPROVED** [review]
Clean code and UI 
Good folder structure 
Good use of semantic tags 
Good for approval

File: src/Components/EditProfile/EditProfile.jsx [hunk 1]
```diff
@@ -38,6 +79,13 @@ export const EditProfile = ({ handleEditModalClose, editModalOpen }) => {
               name="avatar"
               type="file"
               accept="image/*"
+              // onChange={(event) =>
+              //   setUserData((prev) => ({
+              //     ...prev,
+              //     avatar: event.target.files[0],
+              //   }))
+              // }
```
Comment (msdeshmukh009) [inline src/Components/EditProfile/EditProfile.jsx]: You can remove this comment.

File: src/Components/SuggestionsSideBar/SuggestionsSideBar.jsx [hunk 2]
```diff
@@ -7,40 +7,67 @@ import ListItemAvatar from "@mui/material/ListItemAvatar";
...
+
+  // filter user on the basis of user name for follow/ unfollow
+
+  const currentUser = users?.find(
+    (each) => each.username === userInfo.username
+  );
+
+  const currentUserFollowing = currentUser?.following;
+
+  const userSuggestion = [...users].filter(
```
Comment (msdeshmukh009) [inline src/Components/SuggestionsSideBar/SuggestionsSideBar.jsx]: I guess you don't need to make a copy of  `users `. `users.filter()` will return a new array anyway.

**Review by msdeshmukh009 — APPROVED** [review]
- Profile pic is not getting updated in the navbar. 
- I think the user feed should only show posts from followed users.

The rest of the functionalities are working fine.

**Comment by varnita619** [comment]
> * Profile pic is not getting updated in the navbar.
> * I think the user feed should only show posts from followed users.
> 
> The rest of the functionalities are working fine.

done

**Review by Aman-Jain1996 — APPROVED** [review]
Clean and modular code , good to merge

**Pull request merged**